test/conftest.py

```diff
@@ -24,39 +23,38 @@ def multisensor_6_state_fixture():
     return json.loads(load_fixture("multisensor_6_state.json"))
 
 
-@pytest.fixture(name="client_session")
-def client_session_fixture():
-    """Mock an aiohttp client session."""
-    return AsyncMock(spec_set=ClientSession)
+class MockClient(Client):
+    """Mock client."""
 
+    def __init__(self):
+        """Initialize mock client."""
+        super().__init__("ws://test.org", None)
+        self.state = STATE_CONNECTED
+        self.mock_responses: List[Tuple[dict, dict]] = []
+        self.mock_commands = []
 
-@pytest.fixture(name="ws_client")
-def ws_client_fixture():
-    """Mock a websocket client."""
-    return AsyncMock(spec_set=ClientWebSocketResponse)
+    async def async_send_command(self, message):
+        """Send a command and get a response."""
+        for match_command, response in self.mock_responses:
+            if all(message[key] == value for key, value in match_command.items()):
+                self.mock_commands.append(message)
+                return response
 
+        raise RuntimeError("Command not mocked!")
 
-@pytest.fixture(name="uuid4")
-def mock_uuid_fixture():
-    """Patch uuid4."""
-    uuid4_hex = "1234"
-    with patch("uuid.uuid4") as uuid4:
-        uuid4.return_value.hex = uuid4_hex
-        yield uuid4_hex
+    async def async_send_json_message(self, message):
+        """Send JSON message."""
+        raise RuntimeError("Method not mocked!")
 
+    def mock_command(self, match_command: dict, response: dict):
```

Comment:
I'd create a factory fixture that return a function that does the mocking, setting return values etc.

Reply:
Factory fixture? It's not a fixture but a mocked command here.

Reply:
I mean a pytest fixture function that returns a function that we can call to apply the mock command and return value.